noxfile.py

```diff
@@ -59,37 +56,30 @@ def get_requirements_filepath(
 
 @nox.session
 def requirements(session):
-    """Regenerate pinned requirements files used during CI."""
+    """Regenerate pinned requirements files."""
 
     session.install("uv >= 0.1.39")
 
-    category_version_resolution = [
-        ("all", maxpython, "highest"),
-        ("docs", maxpython, "highest"),
-        ("tests", minpython, "lowest-direct"),
+    category_version_resolution: list[tuple[str, str, str]] = [
+        ("tests", version, resolution)
+        for version in supported_python_versions
+        for resolution in ("highest", "lowest-direct")
```

Comment:
Prior to this change, a requirements file was only created for the `lowest-direct` option for Python 3.10.  With this change, this is now done for all supported versions of Python.  This change will enable us to test PlasmaPy against the lowest versions of direct dependencies for all versions of Python (in our weekly tests).